### ISSUE TITLE
Color tint in screenshare when using Wayland

### ISSUE THREAD

**Issue by moka491** [maltejur/discord-screenaudio#116]
Hey, I've got a weird issue today where screensharing anything on Plasma + Wayland
will have a weird color shift to it, I think it's actually an exact color negative of the original image.

![image](https://user-images.githubusercontent.com/31903525/222240522-ac777528-5cd1-4522-a83f-990626a3add1.png)
On the right you see MPV playing a video, on the left it's discord-screenaudio sharing it.
I tried various sources, not just MPV but also sharing the whole desktop, or a game.

If I switch to an Xorg Plasma session, the problem does not occur.

It also didn't occur yesterday (on Wayland) so I assume the problem came with an update to some other packages (Kernel 6.2? new xdg portal version or something? I can't check unfortunately)

Using an older 1.6.3 of discord-screenaudio didn't help (I tried the flatpak).
Everything else works great, so I hope we can sort this out together.

**Comment by 0xAnansi** [comment]
Can confirm on Garuda (Arch based) Wayland, blue and red are inverted. Green is untouched.

The packages that were upgraded before it started bugging out are the following:

```
[2023-02-28T16:27:54+0100] [ALPM] upgraded audit (3.0.9-2 -> 3.1-1)
[2023-02-28T16:27:54+0100] [ALPM] upgraded libjpeg-turbo (2.1.5-2 -> 2.1.5.1-1)
[2023-02-28T16:27:54+0100] [ALPM] upgraded libwacom (2.6.0-1 -> 2.6.0-2)
[2023-02-28T16:27:55+0100] [ALPM] upgraded iso-codes (4.12.0-1 -> 4.13.0-1)
[2023-02-28T16:27:55+0100] [ALPM] upgraded bluedevil (1:5.27.1-1 -> 1:5.27.2-1)
[2023-02-28T16:27:55+0100] [ALPM] upgraded kdecoration (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:27:55+0100] [ALPM] upgraded breeze (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:27:55+0100] [ALPM] upgraded fluidsynth (2.3.1-1 -> 2.3.1-2)
[2023-02-28T16:27:55+0100] [ALPM] upgraded starship (1.13.0-1 -> 1.13.1-1)
[2023-02-28T16:27:55+0100] [ALPM] upgraded garuda-fish-config (1.5.11-1 -> 1.5.12-1)
[2023-02-28T16:27:57+0100] [ALPM] upgraded sweet-theme-full-git (r305.5fda89d-1 -> r307.8028577-1)
[2023-02-28T16:27:57+0100] [ALPM] upgraded libass (0.17.0-1 -> 0.17.1-1)
[2023-02-28T16:27:57+0100] [ALPM] upgraded libksysguard (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:27:57+0100] [ALPM] upgraded ksystemstats (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:27:57+0100] [ALPM] upgraded kactivitymanagerd (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:27:57+0100] [ALPM] upgraded kde-cli-tools (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:27:57+0100] [ALPM] upgraded milou (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:27:57+0100] [ALPM] upgraded layer-shell-qt (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:27:57+0100] [ALPM] upgraded libkscreen (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:27:57+0100] [ALPM] upgraded kscreenlocker (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:27:58+0100] [ALPM] upgraded kwin (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:27:58+0100] [ALPM] upgraded xdg-desktop-portal-kde (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:27:58+0100] [ALPM] upgraded plasma-integration (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:27:58+0100] [ALPM] upgraded kpipewire (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:27:58+0100] [ALPM] upgraded oxygen-sounds (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:28:04+0100] [ALPM] upgraded plasma-workspace (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:28:04+0100] [ALPM] upgraded garuda-dr460nized (3.0.1-1 -> 3.0.2-1)
[2023-02-28T16:28:07+0100] [ALPM] upgraded kde-gtk-config (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:28:07+0100] [ALPM] upgraded kdeplasma-addons (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:28:07+0100] [ALPM] upgraded khotkeys (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:28:07+0100] [ALPM] upgraded systemsettings (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:28:07+0100] [ALPM] upgraded kinfocenter (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:28:07+0100] [ALPM] upgraded kmenuedit (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:28:07+0100] [ALPM] upgraded kscreen (5.27.1.1-1 -> 5.27.2-1)
[2023-02-28T16:28:07+0100] [ALPM] upgraded kwallet-pam (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:28:07+0100] [ALPM] upgraded kwayland-integration (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:28:07+0100] [ALPM] upgraded lib32-fluidsynth (2.3.1-1 -> 2.3.1-2)
[2023-02-28T16:28:07+0100] [ALPM] upgraded lib32-libjpeg-turbo (2.1.5-4 -> 2.1.5.1-1)
[2023-02-28T16:28:20+0100] [ALPM] upgraded linux-zen (6.1.12.zen1-1 -> 6.2.1.zen1-1)
[2023-02-28T16:28:25+0100] [ALPM] upgraded linux-zen-headers (6.1.12.zen1-1 -> 6.2.1.zen1-1)
[2023-02-28T16:28:31+0100] [ALPM] upgraded logseq-desktop (0.8.17-1 -> 0.8.17-3)
[2023-02-28T16:28:31+0100] [ALPM] upgraded make (4.4-1 -> 4.4.1-1)
[2023-02-28T16:28:31+0100] [ALPM] upgraded openfortivpn (1.19.0-2 -> 1.20.1-1)
[2023-02-28T16:28:31+0100] [ALPM] upgraded openmpi (4.1.4-4 -> 4.1.5-1)
[2023-02-28T16:28:31+0100] [ALPM] upgraded pandoc-crossref (0.3.15.0-8 -> 0.3.15.1-1)
[2023-02-28T16:28:31+0100] [ALPM] upgraded plasma-browser-integration (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:28:32+0100] [ALPM] upgraded polkit-kde-agent (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:28:32+0100] [ALPM] upgraded powerdevil (5.27.1-2 -> 5.27.2-1)
[2023-02-28T16:28:36+0100] [ALPM] upgraded plasma-desktop (5.27.1-2 -> 5.27.2-1)
[2023-02-28T16:28:37+0100] [ALPM] upgraded plasma-firewall (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:28:38+0100] [ALPM] upgraded plasma-nm (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:28:38+0100] [ALPM] upgraded plasma-pa (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:28:38+0100] [ALPM] upgraded plasma-systemmonitor (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:28:38+0100] [ALPM] upgraded plasma-thunderbolt (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:28:38+0100] [ALPM] upgraded plasma-wayland-session (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:28:38+0100] [ALPM] upgraded python-pooch (1.6.0-5 -> 1.7.0-1)
[2023-02-28T16:28:39+0100] [ALPM] upgraded python-stem (1.8.0-6 -> 1.8.1-1)
[2023-02-28T16:28:39+0100] [ALPM] upgraded sddm-kcm (5.27.1-1 -> 5.27.2-1)
[2023-02-28T16:28:39+0100] [ALPM] upgraded sudo (1.9.13-1 -> 1.9.13.p1-1)
```

**Comment by Schmene** [comment]
I am also experiencing this issue on Fedora 37. Started noticing it after upgrading to plasma 5.27.2

**Comment by maltejur** [comment]
https://bugs.kde.org/show_bug.cgi?id=466655

**Comment by TheKarion** [comment]
Happening to me as well on EndeavourOS + KDE Plasma + Wayland + Hybrid Nvidia Setup

**Comment by regunakyle** [comment]
> https://bugs.kde.org/show_bug.cgi?id=466655

The issue is fixed in KDE 5.27.3 according to the issue log. 

(I am using 5.27.2 though, so I still have this issue)

**Comment by moka491** [comment]
Can confirm it is fixed on 5.27.3. Thanks for the help everyone!
